tests/unit/test_remote_write.py

```diff
@@ -153,6 +153,63 @@ def test_no_remote_write_endpoint_provided(self):
         self.harness.update_relation_data(rel_id, "provider/0", {})
         assert list(self.harness.charm.remote_write_consumer.endpoints) == []
 
+    def test_alert_rule_is_correct(self):
+        rel_id = self.harness.add_relation(RELATION_NAME, "provider")
+        self.harness.add_relation_unit(rel_id, "provider/0")
+        rules = json.loads(
+            self.harness.get_relation_data(rel_id, self.harness.charm.app)["alert_rules"]
+        )
+        for group in rules["groups"]:
+            if group["name"].endswith("with_template_string_alerts"):
+                expr = group["rules"][0]["expr"]
+                self.assertIn("juju_model", expr)
+                self.assertIn("juju_model_uuid", expr)
+                self.assertIn("juju_application", expr)
+                self.assertIn("juju_charm", expr)
+                self.assertNotIn("juju_unit", expr)
+                self.assertEqual(
+                    set(group["rules"][0]["labels"]),
+                    {
+                        "juju_application",
+                        "juju_charm",
+                        "juju_model",
+                        "juju_model_uuid",
+                        "severity",
+                    },
+                )
+                break
+        else:
+            assert False  # Could not find the correct alert rule to check
+
+    def test_alert_rule_is_correct_with_unit(self):
+        rel_id = self.harness.add_relation(RELATION_NAME, "provider")
+        self.harness.add_relation_unit(rel_id, "provider/0")
+        rules = json.loads(
+            self.harness.get_relation_data(rel_id, self.harness.charm.app)["alert_rules"]
+        )
+        for group in rules["groups"]:
+            if group["name"].endswith("with_template_string_and_unit_alerts"):
+                expr = group["rules"][0]["expr"]
+                self.assertIn("juju_model", expr)
+                self.assertIn("juju_model_uuid", expr)
+                self.assertIn("juju_application", expr)
+                self.assertIn("juju_charm", expr)
+                self.assertIn("juju_unit", expr)
+                self.assertEqual(
+                    set(group["rules"][0]["labels"]),
+                    {
+                        "juju_application",
+                        "juju_charm",
+                        "juju_model",
+                        "juju_model_uuid",
+                        "severity",
+                        "juju_unit",
+                    },
+                )
+                break
+        else:
+            assert False  # Could not find the correct alert rule to check
```

Comment:
Maybe instead of for-else it would be clearer if you first `filter(str.endswith, ...)`?

Reply:
I find this way to be more clear. If I'm outvoted, I'll change it

Reply:
The `else` is paired with the `for` not the `if`